### PR TITLE
Valhalla test updates and disable failing tests

### PR DESCRIPTION
**Pull request by theresa-m**
After this change set the Valhalla more tests should be passing. Tests that are not passing are disabled for now.

This change should be merged after https://github.com/eclipse-openj9/openj9/pull/19646

Changes include:
- removing unused code from ValueTypeTests including the myUnsafe field
- Replace q signatures with L. Mark all generated value fields as NullRestricted. Previously all value classes were marked as primitive, this most closely matches the behavior.
- Remove and modify a few tests in ValueTypeTests, disable remaining

The following issues and comments have been created to track the excluded tests:
- https://github.com/eclipse-openj9/openj9/issues/19692
- https://github.com/eclipse-openj9/openj9/issues/19460#issuecomment-2163847738
- https://github.com/eclipse-openj9/openj9/issues/19677
- https://github.com/eclipse-openj9/openj9/issues/19708

### CODE REVIEW

**Comment by hangshao0** [comment]
@theresa-m Please rebase. There is a merge conflict. Also the commit [Remove CFR_ACC_VALUE_TYPE and CFR_ACC_PRIMITIVE_VALUE_TYPE](https://github.com/eclipse-openj9/openj9/pull/19690/commits/52ce50b7031c6b7ab23b6529c49ba88d9cd042c8) is already reviewed and merged in https://github.com/eclipse-openj9/openj9/pull/19631. It should not appear in this PR again.

**Comment by theresa-m** [comment]
I have rebased this change.

File: test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeGenerator.java [hunk 1]
```diff
@@ -514,7 +521,7 @@ private static void testCheckCastOnInvalidQtype(ClassWriter cw) {
 		MethodVisitor mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC, "testCheckCastOnInvalidQtype", "()Ljava/lang/Object;", null, null);
 		mv.visitCode();
 		mv.visitInsn(ACONST_NULL);
-		mv.visitTypeInsn(CHECKCAST, "QClassDoesNotExist;");
+		mv.visitTypeInsn(CHECKCAST, "LClassDoesNotExist;");
```
Comment (hangshao0) [inline test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeGenerator.java:524]: As Q type is being replaced by L type, the name of this test `testCheckCastOnInvalidQtype` can be updated as well.

File: test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java [hunk 2]
```diff
@@ -195,30 +193,30 @@ public class ValueTypeTests {
 
 	/* fields */
 	static String typeWithSingleAlignmentFields[] = {
-		"tri:QTriangle2D;:value",
-		"point:QPoint2D;:value",
-		"line:QFlattenedLine2D;:value",
-		"i:QValueInt;:value",
-		"f:QValueFloat;:value",
-		"tri2:QTriangle2D;:value"
+		"tri:LTriangle2D;:NR",
```
Comment (hangshao0) [inline test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java:196]: Not sure if it is obvious to everyone that `NR` means `Null Restricted`. It is better to make a comment about this in the first use of `NR` in this file.

File: test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java [hunk 3]
```diff
@@ -327,7 +314,7 @@ static public void testCreateArrayPoint2D() throws Throwable {
 		assertEquals(getY.invoke(point2D_2_check), getY.invoke(point2D_2));
 	}
 
-	@Test(priority=5)
+	@Test(priority=5, enabled = false)
```
Comment (hangshao0) [inline test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java:317]: For the tests in this file disabled by this PR, could you add a comment with the issue number tacking the failure ?

File: test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeGenerator.java (hunk 1 above)
Comment (theresa-m) [reply test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeGenerator.java:524]: Actually I think this test can be removed entirely, testCheckCastOnInvalidLtype already exists.

**Comment by hangshao0** [comment]
Jenkins test sanity,extended zlinuxval jdknext

**Comment by hangshao0** [comment]
Jenkins test sanity,extended winval jdknext

**Comment by hangshao0** [comment]
The Windows [build](https://openj9-jenkins.osuosl.org/job/Build_JDKnext_x86-64_windows_valhalla_Personal/42/) failed due to machine issue:
`11:16:56  ERROR: Cannot delete workspace :Unable to delete 'F:\Users\jenkins\workspace\Build_JDKnext_x86-64_windows_valhalla_Personal\openssl\NUL'`

> Tests that are not passing are disabled for now.

I was assuming all tests should pass. Could you look at the failures in the Linux s390 build ? @theresa-m

**Comment by theresa-m** [comment]
> The Windows [build](https://openj9-jenkins.osuosl.org/job/Build_JDKnext_x86-64_windows_valhalla_Personal/42/) failed due to machine issue: `11:16:56 ERROR: Cannot delete workspace :Unable to delete 'F:\Users\jenkins\workspace\Build_JDKnext_x86-64_windows_valhalla_Personal\openssl\NUL'`
> 
> > Tests that are not passing are disabled for now.
> 
> I was assuming all tests should pass. Could you look at the failures in the Linux s390 build ? @theresa-m

Hmm yea I'll check it out. I was mostly verifying that on my mac which is not able to run all the variations.

**Comment by hangshao0** [comment]
The PR build link on Linux s390 is https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJ9/5637/

File: test/functional/Valhalla/playlist.xml [hunk 4]
```diff
@@ -63,13 +63,13 @@
 		<variations>
 			<variation>-Xjit:count=0</variation>
 			<variation>-Xjit:count=1,disableAsyncCompilation -Xgcpolicy:optthruput</variation>
-			<variation>-Xjit:count=1,disableAsyncCompilation -Xgcpolicy:optthruput -XX:ValueTypeFlatteningThreshold=99999 -XX:-EnableArrayFlattening</variation>
+			<!--<variation>-Xjit:count=1,disableAsyncCompilation -Xgcpolicy:optthruput -XX:ValueTypeFlatteningThreshold=99999 -XX:-EnableArrayFlattening</variation>
```
Comment (hangshao0) [inline test/functional/Valhalla/playlist.xml:66]: Please include the JIT issue number as a comment here.

Comment (theresa-m) [reply test/functional/Valhalla/playlist.xml:67]: Its there now.

**Comment by theresa-m** [comment]
I have disabled the failing tests and updated the issues for excluded tests. Can you try the build again @hangshao0 ?

File: test/functional/Valhalla/playlist.xml (hunk 4 above)
Comment (hangshao0) [reply test/functional/Valhalla/playlist.xml:67]: FYI @hzongaro @a7ehuo Some JIT variations of this test are being disabled because of so many failures.

**Comment by hangshao0** [comment]
Jenkins test sanity,extended alinuxval jdknext

**Review by hangshao0 — APPROVED** [review]

**Pull request merged**